Framework/PythonInterface/plugins/algorithms/IntegratePeaksShoeboxTOF.py

```diff
@@ -444,7 +447,28 @@ def PyExec(self):
                 )
                 x, y, esq, ispecs = get_and_clip_data_arrays(ws, peak_data, pk_tof, kernel, nshoebox)
                 # integrate at previously found ipos
-                ipos = [*np.argwhere(ispecs == weak_pk.ispec)[0], np.argmin(abs(x - weak_pk.tof))]
+                if weak_pk.ispec in ispecs:
```

Comment:
Think this could just be this?
```
if weak_pk.ispec not in ispecs:
    nrows = max(kernel.shape[0], weak_pk.kernel_shape[0])
    ncols = max(kernel.shape[0], weak_pk.kernel_shape[0])
    peak_data = array_converter.get_peak_data(
                        peak,
                        peak.getDetectorID(),
                        bank_name,
                        nshoebox * kernal_nrows_weak_pk,
                        nshoebox * kernal_ncols_weak_pk,
                        nrows_edge,
                        ncols_edge,
                    )
    x, y, esq, ispecs = get_and_clip_data_arrays(ws, peak_data, pk_tof, kernel, nshoebox)
ipos = [*np.argwhere(ispecs == weak_pk.ispec)[0], np.argmin(abs(x - weak_pk.tof))]